### DISCUSSION TITLE
Building and compiling custom Simpleperf

### DISCUSSION THREAD

**Discussion by hosunnyc** [android/ndk#1839] [Q&A] [answered]
Hi, I just wanted to ask and build on this [comment](https://github.com/android/ndk/discussions/1833#discussioncomment-4799917).
How would I go about changing the code for my own custom usage?
I would like to then cross compile for my Pixel 6 (Android 13). I am also doing this on Ubuntu 20.04 WSL2 in case that might have some differences. Thanks!

**Comment by enh-google** [comment, marked as answer]
https://source.android.com/docs/setup/build/building

**Reply by hosunnyc** [reply]
https://android.googlesource.com/platform/ndk/+/master/docs/Building.md
How is this and the link you answered different?

**Reply by enh-google** [reply]
your link is how to build the NDK. my link is how to build the platform. simpleperf is built as part of the platform, and simply packaged as a prebuilt in the NDK. (so the NDK "build" for simpleperf is just a cp.)

**Reply by hosunnyc** [reply]
I see. After downloading the source, setting up the environment, and choosing a target. Is it correct to  understand that to build simpleperf, I would have to navigate into `$SOURCE/system/extras/simpleperf` and run `mma`

**Reply by enh-google** [reply]
correct. or just `mm`. (you probably want `-j` for some parallelism!)

**Reply by pirama-arumuga-nainar** [reply]
The [simpleperf doc](https://android.googlesource.com/platform/system/extras/+/master/simpleperf/doc/README.md#bugs-and-contribution) has an extra step at the end that might be useful.  @yabinc added it recently and I didn't realize its connection to this thread until now.
